### PR TITLE
SURF-1165

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: SURF-1165

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 5cfeb54b7ab87bb9e80f4ad772876cc2e80526d1

https://app.netlify.com/sites/helix-ui/deploys/5bdbd8e6e39e7c1831e33acf

**Comment by cathysiller** [comment]
hey @CITguy, the section "Scrollable Content (configured hx-div)" scrolled in Chrome and Safari and did not scroll for me in Firefox, Edge, and IE :/ the hx-div is supposed to scroll, correct?

**Comment by CITguy** [comment]
@cathysiller, :sigh: yes, it should. The deprecated markup still scrolls, so I'll have to see how the two differ.

File: docs/elements/hx-popover/index.html [hunk 1]
```diff
@@ -76,6 +79,7 @@
 
 {% block properties %}
   <dl>
+    <!-- Positionable -->
```
Comment (100stacks) [inline docs/elements/hx-popover/index.html]: please add `Mixin`

File: docs/styles/demo/tooltip-docs.less [hunk 2]
```diff
@@ -3,8 +3,16 @@
   flex-wrap: wrap;
   z-index: 0;
 
-  hx-tooltip {
+  > hx-tooltip {
     margin: 1.25rem;
     position: relative;
   }
+
+  &.vertical {
```
Comment (100stacks) [inline docs/styles/demo/tooltip-docs.less]: please review

File: src/helix-ui/elements/HXTooltipElement.js [hunk 3]
```diff
@@ -38,173 +27,284 @@ export class HXTooltipElement extends HXElement {
...
 
     /**
-     * Where to position the menu in relation to its reference element.
+     * External element that controls tooltip visibility.
      *
-     * @default "top"
-     * @type {PositionString}
+     * Returns the first element with an "id" matching the tooltip's "for" value.
+     *
+     * @readonly
```
Comment (100stacks) [inline src/helix-ui/elements/HXTooltipElement.js]: please add an `override`

File: src/helix-ui/mixins/Revealable.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        /** @override */
+        $onAttributeChange (attr, oldVal, newVal) {
+            super.$onAttributeChange(attr, oldVal, newVal);
+
+            if (attr === 'open') {
+                let isOpen = (newVal !== null);
+                this.setAttribute('aria-hidden', !isOpen);
+
+                if (isOpen) {
```
Comment (100stacks) [inline src/helix-ui/mixins/Revealable.js]: make this into an ternary please :)

File: src/helix-ui/styles/elements/hx-popover.less [hunk 5]
```diff
@@ -1,75 +1,71 @@
...
+  > hx-div:not([scroll]) {
+    overflow: hidden;
   }
 
-  & > &-body {
-    &:extend(hx-popover > hx-div);
+  > hx-div {
+    flex-grow: 1;
+    max-height: 20rem; //~320px
+    order: 2;
```
Comment (100stacks) [inline src/helix-ui/styles/elements/hx-popover.less]: please remove `order: 2;`

File: src/helix-ui/styles/elements/hx-popover.less [hunk 6]
```diff
@@ -1,75 +1,71 @@
...
+    &:last-child {
+      padding-bottom: 1.25rem;
+    }
   }
 
-  & > &-foot {
-    &:extend(hx-popover > footer all);
+  > footer {
+    flex-shrink: 0;
+    order: 3;
```
Comment (100stacks) [inline src/helix-ui/styles/elements/hx-popover.less]: please remove `order: 3;`

File: src/helix-ui/utils/index.js [hunk 7]
```diff
@@ -86,3 +86,34 @@ export default {
...
+ * @param {class} baseClass - Base class to apply mixin behavior
+ * @returns {MixinBuilder}
+ *
+ * @example
+ * import { mix } from 'utils';
+ *
+ * // Define unique superclass with behaviors from one or more mixin classes
+ * class _ProtoClass extends mix(HXElement).with(MixableClass) {}
+ *
+ * // Extend unique superclass and define additionala logic
```
Comment (100stacks) [inline src/helix-ui/utils/index.js]: fix misspelling

File: src/helix-ui/utils/index.js [hunk 8]
```diff
@@ -86,3 +86,34 @@ export default {
...
+ * class HXNewElement extends _ProtoClass {
+ *   // logic unique to HXNewElement ...
+ * }
+ */
+export function mix (baseClass) {
+    class MixinBuilder {
+        constructor (superclass) {
```
Comment (100stacks) [inline src/helix-ui/utils/index.js]: please review use of constructor

File: src/helix-ui/utils/position/index.js [hunk 9]
```diff
@@ -7,7 +7,7 @@ import Offset, { offsetFunctionMap } from './offset';
  * @typedef {Object} PositionConfig
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/utils/position/index.js]: double check if we can revert changes to this file.

File: src/helix-ui/elements/HXTooltipElement.js (hunk 3 above)
Comment (CITguy) [reply src/helix-ui/elements/HXTooltipElement.js]: Discovered that `@override` isn't needed.

File: src/helix-ui/mixins/Positionable.js [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+ */
+export const Positionable = (superclass) => {
+    class ProtoClass extends mix(superclass).with(Revealable) {}
+
+    /** @lends Positionable */
+    class _Positionable extends ProtoClass {
+        constructor () {
+            super();
+
+            this.reposition = this.reposition.bind(this);
```
[the remark this answers is not in the document]
Comment (CITguy) [reply src/helix-ui/mixins/Positionable.js]: removed

File: docs/components/tooltips/test.html [hunk 11]
```diff
@@ -12,14 +12,9 @@
...
-    <h2>Appearance</h2>
+    <h2>Appearance (simple)</h2>
     <div class="tooltip-appearance">
-      {% for _position in [
-        'top', 'top-left', 'top-right',
-        'bottom', 'bottom-left', 'bottom-right',
-        'right', 'right-top', 'right-bottom',
-        'left', 'left-top', 'left-bottom'
-      ] %}
+      {% for _position in data.positions.withArrow %}
```
[the remark this answers is not in the document]
Comment (CITguy) [reply docs/components/tooltips/test.html]: done

File: src/helix-ui/utils/position/index.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (CITguy) [reply src/helix-ui/utils/position/index.js]: Can't revert. HXMenuElement and HXSearchAssistanceElement both use updated logic.

File: docs/elements/hx-popover/index.html [hunk 12]
```diff
@@ -57,71 +67,104 @@
 {% block attributes %}
   <dl>
     <!-- Revealable Mixin -->
-    <dt>open</dt>
+    <dt>open (optional)</dt>
     <dd>
-      When present, the popover will be revealed.
+      <p>
+        If present, the {{page.title}} will be shown on screen.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 13]
```diff
@@ -57,71 +67,104 @@
...
     <dd>
-      <p class="comfortable">
-        <i>(read-only)</i>
-        Returns the HTML element used as a reference to calculate the position of
-        the popover. If the <code>relative-to</code> attribute is set, this will
-        be the element with ID matching the <code>relativeTo</code> property value.
-        Otherwise, this will be the <code>controlElement</code>.
+      <p>
+        HTML element used as a point of reference for calculating the fixed
+        position of the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 14]
```diff
@@ -57,71 +67,104 @@
...
     </dd>
 
     <!-- Positionable Mixin-->
-    <dt>position</dt>
+    <dt>position (optional)</dt>
     <dd>
-      Configures the position of the popover in relation to its
-      relative element.
+      <p>
+        Configures the position of the {{page.title}}
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 15]
```diff
@@ -57,71 +67,104 @@
...
 
     <!-- Positionable Mixin -->
     <dt>relative-to (optional)</dt>
     <dd>
-      ID of an element that the popover is positioned relative to.
+      <p>
+        The <code>id</code> of an element in the same document as the
+        <code>{{page.title}}</code> element and the <i>control element</i>.
+        The first element with a matching id is known as the
+        <i>relative element</i> for the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 16]
```diff
@@ -57,71 +67,104 @@
...
+      controlElement (HTMLElement)
+      <hx-pill class="hxSubBody">read-only</hx-pill>
+    </dt>
     <dd>
       <p>
-        <i>(read-only)</i>
-        Returns the HTML element with the <code>aria-controls</code>
-        attribute that matches the popover ID.
+        HTML element whose <code>aria-controls</code>
+        attribute matches the id of the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 17]
```diff
@@ -57,71 +67,104 @@
...
+      </p>
+      <p>
+        The control element is used to attach various DOM event listeners,
+        in order to determine when to show or hide the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 18]
```diff
@@ -57,71 +67,104 @@
...
       </p>
     </dd>
 
     <!-- Revealable Mixin -->
     <dt>open (Boolean [false])</dt>
     <dd>
       <p>
-        Manipulates the <code>open</code> attribute.
+        Indicates whether or not the {{page.title}} should be displayed.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-popover/index.html [hunk 19]
```diff
@@ -57,71 +67,104 @@
...
       </p>
     </dd>
 
     <!-- Positionable Mixin-->
     <dt>position (String [bottom-right])</dt>
     <dd>
       <p>
-        Manipulates the <code>position</code> attribute.
+        Configures the position of the {{page.title}}, in relation to its
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 20]
```diff
@@ -37,78 +47,126 @@
...
 {% block attributes %}
   <dl>
+    <!-- HXTooltipElement -->
     <dt>for</dt>
     <dd>
-      When present, the for attribute specifies which element a tooltip is bound to.
+      <p>
+        The <code>id</code> of an element in the same document as the
+        <code>{{page.title}}</code> element. The first element with a
+        matching id is known as the <i>control element</i> for the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 21]
```diff
@@ -37,78 +47,126 @@
...
     </dd>
 
-  <dl>
     <!-- Revealable Mixin -->
-    <dt>open</dt>
+    <dt>open (optional)</dt>
     <dd>
-      When present, the tooltip will be revealed.
+      <p>
+        If present, the {{page.title}} will be shown on screen.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 22]
```diff
@@ -37,78 +47,126 @@
...
     </dd>
 
     <!-- Positionable Mixin-->
-    <dt>position</dt>
+    <dt>position (optional)</dt>
     <dd>
-      Configures the position of the tooltip in relation to its
-      relative element.
+      <p>
+        Configures the position of the {{page.title}}
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 23]
```diff
@@ -37,78 +47,126 @@
...
 
     <!-- Positionable Mixin -->
     <dt>relative-to (optional)</dt>
     <dd>
-      ID of an element that the tooltip is positioned relative to.
+      <p>
+        The <code>id</code> of an element in the same document as the
+        <code>{{page.title}}</code> element and the <i>control element</i>.
+        The first element with a matching id is known as the
+        <i>relative element</i> for the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 24]
```diff
@@ -37,78 +47,126 @@
...
+      </p>
+      <p>
+        If omitted, the <i>control element</i> will act as the
+        relative element.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `relative element` in `<i>`

File: docs/elements/hx-popover/index.html [hunk 25]
```diff
@@ -57,71 +67,104 @@
...
-      ID of an element that the popover is positioned relative to.
+      <p>
+        The <code>id</code> of an element in the same document as the
+        <code>{{page.title}}</code> element and the <i>control element</i>.
+        The first element with a matching id is known as the
+        <i>relative element</i> for the {{page.title}}.
+      </p>
+      <p>
+        If omitted, the <i>control element</i> will act as the
+        relative element.
```
Comment (CITguy) [inline docs/elements/hx-popover/index.html]: wrap `relative element` in `<i>`

File: docs/elements/hx-tooltip/index.html [hunk 26]
```diff
@@ -37,78 +47,126 @@
...
+    <dt>
+      controlElement (HTMLElement)
+      <hx-pill class="hxSubBody">read-only</hx-pill>
+    </dt>
+    <dd>
+      <p>
+        HTML element whose ID matches the <code>htmlFor</code> value.
+      </p>
+      <p>
+        The control element is used to attach various DOM event listeners,
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `control element` in `<i>`

File: docs/elements/hx-tooltip/index.html [hunk 27]
```diff
@@ -37,78 +47,126 @@
...
+        in order to determine when to show or hide the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 28]
```diff
@@ -37,78 +47,126 @@
...
+        Reflects the <code>for</code> attribute.
       </p>
     </dd>
 
     <!-- Revealable Mixin -->
     <dt>open (Boolean [false])</dt>
     <dd>
       <p>
-        Determines whether or not tooltip is shown.
+        Indicates whether or not the {{page.title}} should be displayed.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 29]
```diff
@@ -37,78 +47,126 @@
...
       </p>
     </dd>
 
     <!-- Positionable Mixin-->
     <dt>position (String [top])</dt>
     <dd>
       <p>
-        Determines where tooltip is positioned.
+        Configures the position of the {{page.title}}, in relation to its
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

File: docs/elements/hx-tooltip/index.html [hunk 30]
```diff
@@ -37,78 +47,126 @@
...
     <dd>
-      <p class="comfortable">
-        <i>(read-only)</i>
-        Returns the HTML element used as a reference to calculate the position of
-        the popover. If the <code>relative-to</code> attribute is set, this will
-        be the element with ID matching the <code>relativeTo</code> property value.
-        Otherwise, this will be the <code>controlElement</code>.
+      <p>
+        HTML element used as a point of reference for calculating the fixed
+        position of the {{page.title}}.
```
Comment (CITguy) [inline docs/elements/hx-tooltip/index.html]: wrap `{{page.title}}` in `<code>`

**Pull request merged**